### DISCUSSION TITLE
Version 7.40.0-next.1

### DISCUSSION THREAD

**Discussion by bluebill1049** [react-hook-form/react-hook-form#9424] [General]
👍 Improved async `defaultValues` inference

```diff
const fetchData = async () => {
  await sleep(1000)
  return {
    data: 'test'
  }
}

- useForm<Awaited<ReturnTypeof<typeof fetchData>>>({
+ useForm({
  defaultValues:  fetchData, // will infer the promise return type
})
```

👍 Improved `useForm` resetOptions to keep track of dirty input

```diff
- const { formState: { dirtyFields } } = useForm({
+ useForm({ // dirtyFields subscription is no longer required
  values,
  resetOptions: {
    keepDirtyValues; true, // only non dirty input will receive data from the values update
  }
})
```

🪶 improve `useFormContext` perf by avoiding reference re-create with omit 
🦏 close #9410 improve UX when subscribing to both isValid and `errors` state (#9413)

<hr /><em>This discussion was created from the release <a href='https://github.com/react-hook-form/react-hook-form/releases/tag/v7.40.0-next.1'>Version 7.40.0-next.1</a>.</em>

**Comment by FeMaffezzolli** [comment]
That's awesome. Thanks for the great work! 😃

**Reply by bluebill1049** [reply]
Thank you for the kind words <3
